dashboard/tests/test_dashboard.py

```diff
@@ -222,3 +235,130 @@ def test_http_get(ray_start_with_dashboard):
+    assert test_agent_cls is not None
+
+    def _has_route(route, method, path):
+        if isinstance(route, aiohttp.web.RouteDef):
+            if route.method == method and route.path == path:
+                return True
+        return False
+
+    def _has_static(route, path, prefix):
+        if isinstance(route, aiohttp.web.StaticDef):
+            if route.path == path and route.prefix == prefix:
+                return True
+        return False
+
+    all_routes = dashboard_utils.ClassMethodRouteTable.routes()
+    assert any(_has_route(r, "HEAD", "/test/route_head") for r in all_routes)
+    assert any(_has_route(r, "GET", "/test/route_get") for r in all_routes)
+    assert any(_has_route(r, "POST", "/test/route_post") for r in all_routes)
+    assert any(_has_route(r, "PUT", "/test/route_put") for r in all_routes)
+    assert any(_has_route(r, "PATCH", "/test/route_patch") for r in all_routes)
+    assert any(
+        _has_route(r, "DELETE", "/test/route_delete") for r in all_routes)
+    assert any(_has_route(r, "*", "/test/route_view") for r in all_routes)
+
+    # Test bind()
+    bound_routes = dashboard_utils.ClassMethodRouteTable.bound_routes()
+    assert len(bound_routes) == 0
+    dashboard_utils.ClassMethodRouteTable.bind(
+        test_agent_cls.__new__(test_agent_cls))
+    bound_routes = dashboard_utils.ClassMethodRouteTable.bound_routes()
+    assert any(_has_route(r, "POST", "/test/route_post") for r in bound_routes)
+    assert all(
+        not _has_route(r, "PUT", "/test/route_put") for r in bound_routes)
+
+    # Static def should be in bound routes.
+    routes.static("/test/route_static", "/path")
+    bound_routes = dashboard_utils.ClassMethodRouteTable.bound_routes()
+    assert any(
+        _has_static(r, "/path", "/test/route_static") for r in bound_routes)
+
+    # Test duplicated routes should raise exception.
+    try:
+
+        @routes.get("/test/route_get")
+        def _duplicated_route(req):
+            pass
+
+        raise Exception("Duplicated routes should raise exception.")
+    except Exception as ex:
+        message = str(ex)
+        assert "/test/route_get" in message
+        assert "test_head.py" in message
+
+    # Test exception in handler
+    post_handler = None
+    for r in bound_routes:
+        if _has_route(r, "POST", "/test/route_post"):
+            post_handler = r.handler
+            break
+    assert post_handler is not None
+
+    loop = asyncio.get_event_loop()
+    r = loop.run_until_complete(post_handler())
+    assert r.status == 200
+    resp = json.loads(r.body)
+    assert resp["result"] is False
+    assert "Traceback" in resp["msg"]
+
+
+def test_async_loop_forever(disable_test_module):
+    counter = [0]
+
+    @dashboard_utils.async_loop_forever(interval_seconds=1)
+    async def foo():
+        counter[0] += 1
+        raise Exception("Test exception")
+
+    loop = asyncio.get_event_loop()
+    loop.create_task(foo())
+    loop.call_later(4, loop.stop)
```

Comment:
4 seconds delay might be too long?

Reply:
It's better to set `interval_seconds` to 0.1 and `call_later` 1 seconds. I will fix it.